diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index 65508eb89..14638f268 100644
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -2247,6 +2247,8 @@ int clk_set_rate(struct clk *clk, unsigned long rate)
 	if (!clk)
 		return 0;
 
+	dev_info(clk->dev, "clk_set_rate: %d\n", rate);
+
 	/* prevent racing with updates to the clock topology */
 	clk_prepare_lock();
 
@@ -2290,6 +2292,8 @@ int clk_set_rate_exclusive(struct clk *clk, unsigned long rate)
 	if (!clk)
 		return 0;
 
+	dev_info(clk->dev, "clk_set_rate_exclusive: %d\n", rate);
+
 	/* prevent racing with updates to the clock topology */
 	clk_prepare_lock();
 
@@ -2327,6 +2331,8 @@ int clk_set_rate_range(struct clk *clk, unsigned long min, unsigned long max)
 	if (!clk)
 		return 0;
 
+	dev_info(clk->dev, "clk_set_rate_range: %d => %d\n", min, max);
+
 	trace_clk_set_rate_range(clk->core, min, max);
 
 	if (min > max) {
